src/ucp/core/ucp_request.h

```diff
@@ -50,12 +50,16 @@ typedef void (*ucp_request_callback_t)(ucp_request_t *req);
 
 
 typedef struct ucp_send_state {
-    size_t                        offset;
+    size_t                        offset;  /* Total offset in overall payload. */
     union {
         struct {
             uct_mem_h             memh;
         } contig;
         struct {
+            size_t                iov_offset;     /* Offset in the IOV item */
+            size_t                iovcnt_offset;  /* The IOV item to start copy */
```

Comment:
@yosefe Also, I can't reuse `offset` for the same purpose as `iov_offset` used. Because, the `offset` used in transmit stage selection (first/middle/last) and have to be incremental. The `iov_offset` is related to particular buffer and is not incremental.